src/main/java/beam/agentsim/events/handling/BeamEventsWriterCSV.java

```diff
@@ -103,15 +105,35 @@ protected void writeEvent(Event event) {
     }
 
     private void registerClass(Class cla) {
-        Field[] fields = cla.getFields();
-        for (Field field : fields) {
-            if ((field.getName().startsWith("ATTRIBUTE_") && (eventTypeToLog == null || !field.getName().startsWith("ATTRIBUTE_TYPE"))) ||
-                    (field.getName().startsWith("VERBOSE_") && (eventTypeToLog == null || !field.getName().startsWith("VERBOSE_")))
-                    ) {
-                try {
-                    attributeToColumnIndexMapping.put(field.get(null).toString(), 0);
-                } catch (IllegalArgumentException | IllegalAccessException e) {
-                    e.printStackTrace();
+        // PathTraversalEvent class is from scala, so we have to have special treatment for them
+        // scala's val and var are not actual fields, but methods (getters and setters)
+        if (cla == PathTraversalEvent.class) {
```

Comment:
Should we maybe make this more of a fallthrough, so that we handle future event classes that are scala also? If that seems too expensive, then maybe we add some sort of fuzzy heuristic where we at least try one path over the other?